### ISSUE TITLE
Cancelling payment sheet with custom flow doesn't send error, if customer has saved cards

### ISSUE THREAD

**Issue by ehaycox** [stripe/stripe-react-native#1419]
**Describe the bug**
If you create a payment sheet with custom flow, and initialize it with a customer that has stripe cards, the payment sheet selects a card by default. This means if you cancel out of the payment sheet, the card is already selected, so it doesn't throw a Cancelled error, it counts it as having selected a payment method.

**To Reproduce**
1. Create payment sheet with custom flow
2. When call initPaymentSheet, fill in stripe customer and ephemeral key for a stripe customer that has saved cards.
3. Open the payment sheet, a card should be selected by default
4. Click the cross on the payment sheet, but presentPaymentSheet returns a null error, and non-null paymentOption with default card.

**Expected behavior**
Should return a cancel error, or at least a null paymentOption

**Smartphone (please complete the following information):**

- Android: OnePlus Nord / OS: 11
- IOS: IPhone 11 / IOS: 16.5

**Comment by remonh87** [comment]
I was fixing an issue in our flutter repo and I can confirm above scenario.

Furthermore there is an inconsistency in propagating result objects between android and iOS. For the following scenario:

1. init payment sheet and after that use present payment sheet (no user so it will only show the Apple Pay or google pay option )
2. When choosing select payment method press cancel.
3. Behaviour android: it provides an error with cancelled.
  Behaviour iOS : it defaults to the Apple Pay payment option and we can continue.

**Comment by ChurikiTenna** [comment]
1+

When will this be fixed??

**Comment by paul-testec** [comment]
any update bro?

**Comment by rjsrikureja** [comment]
can we get an update on this?

**Comment by marlene89** [comment]
+1 getting same issue. When using custom flow, pressing out of the payment sheet or clicking on X button to close it, returns selected default payment option and does not throw any error.

**Comment by amin79** [comment]
any update?

**Comment by Resource-4** [comment]
Facing the same issue in react-native stripe sdk, any update on this? or any work around for this?

**Comment by rjsrikureja** [comment]
I had to switch from custom flow due to this error. I recommend other people do the same if it’s possible in your design.

**Comment by porter-stripe** [comment]
Hi all, is this still an issue that you are expericing?

**Comment by lautaroml** [comment]
> > Hi all, is this still an issue that you are expericing?
> 
> yes, experiencing it in the last version! Is there a chance to fix this soon [@porter-stripe](https://github.com/porter-stripe) ?

Same here! 

@porter-stripe any update on this?

We had to switch from custom flow because of this.

**Comment by porter-stripe** [comment]
Hi all, this is currently intended behavior. If the user has a selected saved card, and hits the "X" we keep that card as selected. The "X" is more of a "close the sheet", rather than "deselect and close". Maybe we should reconsider this behavior. Can you please share the use-case/requirement for wanting it to be cleared and/or throw an error? Why is having it remain selected problematic?

Thanks!

**Comment by ksairi** [comment]
@porter-stripe yes, exactly. I think it should behave the same as the normal flow, if the X is tapped, the promise returns an **error**. It is not happening in the custom flow, hence this open issue, otherwise there is no control on our side to know that  the user closed the sheet instead of continuing with the flow

**Comment by lamhong-playon** [comment]
+1 Getting same issue.
In the Cash App custom payment flow, if the user closes the payment sheet, the app still redirects to Cash App payment. This happens because we cannot detect the dismissal from:

const { error } = await presentPaymentSheet({ timeout: getPaymentSheetTimeout() });

error is always undefined, which causes the app to continue with the next steps instead of treating it as a canceled payment.
